src/dotenv.cr

```diff
@@ -172,10 +173,69 @@ module Dotenv
   # ```
   def load(hash : Hash(String, String)) : Hash(String, String)
     hash.each do |key, value|
-      unless ENV.has_key?(key)
+      unless ENV.has_key?(key) && skip_duplicate_keys
         ENV[key] = value
       end
     end
     hash
   end
+
+  # Allows for overriding existing `ENV` keys. See `load`.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # Dotenv.load ".env"       # => {"VAR" => "Hello"}
+  # Dotenv.load! ".env.test" # => {"VAR" => "World"}
+  # ```
+  def load!(filename : Path | String = ".env") : Hash(String, String)
+    load_with_temp_override do
+      load filename
+    end
+  end
+
+  # Allows for overriding existing `ENV` keys. See `load`.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # Dotenv.load IO::Memory.new("VAR=Hello")  # => {"VAR" => "Hello"}
+  # Dotenv.load! IO::Memory.new("VAR=World") # => {"VAR" => "World"}
+  # ```
+  def load!(io : IO) : Hash(String, String)
+    load_with_temp_override do
+      load io
+    end
+  end
+
+  # Allows for overriding existing `ENV` keys. See `load`.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # Dotenv.load({"VAR" => "Hello"})  # => ENV["VAR"] == "Hello"
+  # Dotenv.load!({"VAR" => "World"}) # => ENV["VAR"] == "World"
+  # ```
+  def load!(hash : Hash(String, String)) : Hash(String, String)
+    load_with_temp_override do
+      load hash
+    end
+  end
+
+  # Allows duplicate `ENV` keys to be overriden. Lock is placed back
+  # on after block finshes. returns the loaded env keys.
+  #
+  # ```
+  # require "dotenv"
+  #
+  # Dotenv.load_with_temp_override do
+  #   Dotenv.load({"VAR" => "test"})
+  # end
+  # ```
+  def load_with_temp_override : Hash(String, String)
+    self.skip_duplicate_keys = false
+    hash = yield
+    self.skip_duplicate_keys = true
+    hash
```

Comment:
All of the `load!` methods just call down to this to set allowing overrides, then return back so it's only overridden when you call `load!`.

Reply:
Could it yield self, so one can can do `Dotenv.load_with_temp_override &.load({"VAR" => "test"})`?
The example will still work

Reply:
I think it's weird, because we can use this methods plus the `load!` one.
Either this methods goes, or the `load!`, but not having both.